1. Comparsion between [#CG methylation/# CG for which you have 4x coverage] AND [#CG methylation/# CG for which you have 4x coverage]/# CG in this 20 bp bin

2. Emailed Denis about which test we should be using for nonparamtic anova, and posthoc test along with correcting for multiple testing... : The Kruskal-Wallis test should be enough, you do not need to use the Wilcoxon/Spearman test unless you want to find exactly which pairs are statistically different. Also, you do not need to correct for multiple comparisons when using Kruskal-Wallis test (or ANOVA). 
FYI, if you decided to use the Wilcoxon test, then the correction for multiple comparisons would be necessary, since you would need to run multiple tests between the expression quartiles.

3. sliding window anova... of 20 bins (100bp regions) for R,V,N,S to determine which region is contributng more (SEE correlation plot at end of CG section)

4. Extract genes from these regions and run go analysis?! 


NOTES
=====

1. Fix alignment of data... DONE (/Users/gturco/Documents/code/Brady/bs_seq_analysis/genic_analysis/reformat_genedata.py)
2. Change this in anova script... DONE
3. run anova script on all plots make heatmap at bottom of each section... have third column for RPKM or Tissue.. DONE
4. add to this...... DONE
5. Make another correlation graph for bigger data similar to Siobhans for Peter: .... DONE



CG Methylation RPKM 1->4
========================================================

#CG methylation/# CG for which you have 4x coverage


```{r fig.width=7, fig.height=5}
library(ggplot2)
library(grid)

file_name = "/Users/gturco/Documents/Data/Sorg/TSS/rpkm_new/tissue/Tissue/CG_all_1.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("root","shoot","vascular","nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8")) + ylim(14,84)
```


Are these changes significant

```{r}
library(PMCMR)
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```

[#CG methylation/# CG for which you have 4x coverage]/# CG in this 20 bp bin

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/11_25_2015/dist_data/TSS_cs_plots/Tissue/CG/CG_1.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("Root","Shoot","Vascular","Nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8"))
```

```{r}
library(PMCMR)
attach(x)
kruskal.test(freq ~ mtype, data = x) 
genebody = x[which(pos > 0 & pos < 1000),]
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=genebody, dist="Tukey")

```




```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/TSS/rpkm_new/tissue/Tissue/CG_all_2.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("root","shoot","vascular","nonvascular"))
ggplot(data=x, aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8")) + ylim(14,84)
```

```{r fig.width=8.5, fig.height=1}

file_name = "/Users/gturco/Documents/code/Brady/bs_seq_analysis/genic_analysis/sig_heatmaps/CG_all_2.txt"
y = read.table(file_name ,sep='\t',header=TRUE)
colnames(y) <- c("pos","sig","mtype")
y$pval <- cut(y$sig, breaks = c(-Inf, 0.001, 0.005, 0.05, 1))
ggplot(y, aes(pos, "", fill = pval)) + geom_tile() + scale_fill_brewer(palette = "Greens",direction=-1)  +
  theme_classic() + theme(text=element_text(size=20))    +  
  theme(text=element_text(size=7), panel.margin = unit(0, "cm"), 
        plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), 
        axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + 
  xlab(NULL) + ylab(NULL) 
  ```
  


Are these changes significant

```{r}
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```

[#CG methylation/# CG for which you have 4x coverage]/# CG in this 20 bp bin

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/11_25_2015/dist_data/TSS_cs_plots/Tissue/CG/CG_2.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("Root","Shoot","Vascular","Nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8"))
```

```{r}
library(PMCMR)
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/TSS/rpkm_new/tissue/Tissue/CG_all_3.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("root","shoot","vascular","nonvascular"))
ggplot(data=x, aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8")) + ylim(14,84)
```


Are these changes significant

```{r}
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```

[#CG methylation/# CG for which you have 4x coverage]/# CG in this 20 bp bin

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/11_25_2015/dist_data/TSS_cs_plots/Tissue/CG/CG_3.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("Root","Shoot","Vascular","Nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8"))
```

```{r}
library(PMCMR)
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/TSS/rpkm_new/tissue/Tissue/CG_all_4.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("root","shoot","vascular","nonvascular"))
ggplot(data=x, aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8")) + ylim(14,84)
```

```{r fig.width=8.5, fig.height=1}

file_name = "/Users/gturco/Documents/code/Brady/bs_seq_analysis/genic_analysis/sig_heatmaps/CG_all_4.txt"
y = read.table(file_name ,sep='\t',header=TRUE)
colnames(y) <- c("pos","sig","mtype")
y$pval <- cut(y$sig, breaks = c(-Inf, 0.001, 0.005, 0.05, 1))
ggplot(y, aes(pos, "", fill = pval)) + geom_tile() + scale_fill_brewer(palette = "Greens",direction=-1)  +
  theme_classic() + theme(text=element_text(size=20))    +  
  theme(text=element_text(size=7), panel.margin = unit(0, "cm"), 
        plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), 
        axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + 
  xlab(NULL) + ylab(NULL) 
  ```
  





Are these changes significant

```{r}
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```

[#CG methylation/# CG for which you have 4x coverage]/# CG in this 20 bp bin

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/11_25_2015/dist_data/TSS_cs_plots/Tissue/CG/CG_4.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("Root","Shoot","Vascular","Nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8"))
```

```{r}
library(PMCMR)
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


```{r fig.width=8.5, fig.height=4}

file_name = "/Users/gturco/Documents/code/Brady/bs_seq_analysis/genic_analysis/sig_heatmaps/CG_all.txt"
y = read.table(file_name ,sep='\t',header=TRUE)
colnames(y) <- c("pos","sig","mtype")
y$pval <- cut(y$sig, breaks = c(-Inf, 0.001, 0.005, 0.05, 1))
ggplot(y, aes(pos, mtype, fill = pval)) + geom_tile() + scale_fill_brewer(palette = "Greens",direction=-1)  +
  theme_classic() + theme(text=element_text(size=20))    +  
  theme(text=element_text(size=12), panel.margin = unit(0, "cm"), 
        plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), 
        axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + 
  xlab(NULL) + ylab(NULL) 
  ```
  
```{r fig.width=8.5, fig.height=4}

file_name = "/Users/gturco/Documents/code/Brady/bs_seq_analysis/genic_analysis/sig_heatmaps/strict/CG_all.txt"
y = read.table(file_name ,sep='\t',header=TRUE)
colnames(y) <- c("pos","sig","mtype")
y$pval <- cut(y$sig, breaks = c(-Inf, 0.001, 0.005, 0.05, 1))
ggplot(y, aes(pos, mtype, fill = pval)) + geom_tile() + scale_fill_brewer(palette = "Greens",direction=-1)  +
  theme_classic() + theme(text=element_text(size=20))    +  
  theme(text=element_text(size=12), panel.margin = unit(0, "cm"), 
        plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), 
        axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + 
  xlab(NULL) + ylab(NULL) 
  ```







CHG Methylation RPKM 1->4
========================================================

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/TSS/rpkm_new/tissue/Tissue/CHG_all_1.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("root","shoot","vascular","nonvascular"))
ggplot(data=x, aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8")) + ylim(5,35)
```


Are these changes significant

```{r}
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


[#CG methylation/# CG for which you have 4x coverage]/# CG in this 20 bp bin

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/11_25_2015/dist_data/TSS_cs_plots/Tissue/CHG/CHG_1.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("Root","Shoot","Vascular","Nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8"))
```

```{r}
library(PMCMR)
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/TSS/rpkm_new/tissue/Tissue/CHG_all_2.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("root","shoot","vascular","nonvascular"))
ggplot(data=x, aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8")) + ylim(5,35)
```


Are these changes significant

```{r}
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


[#CG methylation/# CG for which you have 4x coverage]/# CG in this 20 bp bin

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/11_25_2015/dist_data/TSS_cs_plots/Tissue/CHG/CHG_2.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("Root","Shoot","Vascular","Nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8"))
```

```{r}
library(PMCMR)
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/TSS/rpkm_new/tissue/Tissue/CHG_all_3.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("root","shoot","vascular","nonvascular"))
ggplot(data=x, aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8")) + ylim(5,35)
```


Are these changes significant

```{r}
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


[#CG methylation/# CG for which you have 4x coverage]/# CG in this 20 bp bin

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/11_25_2015/dist_data/TSS_cs_plots/Tissue/CHG/CHG_3.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("Root","Shoot","Vascular","Nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8"))
```

```{r}
library(PMCMR)
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/TSS/rpkm_new/tissue/Tissue/CHG_all_4.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("root","shoot","vascular","nonvascular"))
ggplot(data=x, aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8")) + ylim(5,35)
```


Are these changes significant

```{r}
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


[#CG methylation/# CG for which you have 4x coverage]/# CG in this 20 bp bin

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/11_25_2015/dist_data/TSS_cs_plots/Tissue/CHG/CHG_4.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("Root","Shoot","Vascular","Nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8"))
```

```{r}
library(PMCMR)
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


```{r fig.width=8.5, fig.height=4}

file_name = "/Users/gturco/Documents/code/Brady/bs_seq_analysis/genic_analysis/sig_heatmaps/CHG_all.txt"
y = read.table(file_name ,sep='\t',header=TRUE)
colnames(y) <- c("pos","sig","mtype")
y$pval <- cut(y$sig, breaks = c(-Inf, 0.001, 0.005, 0.05, 1))
ggplot(y, aes(pos, mtype, fill = pval)) + geom_tile() + scale_fill_brewer(palette = "Greens",direction=-1)  +
  theme_classic() + theme(text=element_text(size=20))    +  
  theme(text=element_text(size=12), panel.margin = unit(0, "cm"), 
        plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), 
        axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + 
  xlab(NULL) + ylab(NULL) 
  ```
  
```{r fig.width=8.5, fig.height=4}

file_name = "/Users/gturco/Documents/code/Brady/bs_seq_analysis/genic_analysis/sig_heatmaps/strict/CHG_all.txt"
y = read.table(file_name ,sep='\t',header=TRUE)
colnames(y) <- c("pos","sig","mtype")
y$pval <- cut(y$sig, breaks = c(-Inf, 0.001, 0.005, 0.05, 1))
ggplot(y, aes(pos, mtype, fill = pval)) + geom_tile() + scale_fill_brewer(palette = "Greens",direction=-1)  +
  theme_classic() + theme(text=element_text(size=20))    +  
  theme(text=element_text(size=12), panel.margin = unit(0, "cm"), 
        plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), 
        axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + 
  xlab(NULL) + ylab(NULL) 
  ```




CHH Methylation RPKM 1->4
========================================================


```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/TSS/rpkm_new/tissue/Tissue/CHH_all_1.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("root","shoot","vascular","nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8")) + ylim(5,21)

```


Are these changes significant

```{r}
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```

[#CG methylation/# CG for which you have 4x coverage]/# CG in this 20 bp bin

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/11_25_2015/dist_data/TSS_cs_plots/Tissue/CHH/CHH_1.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("Root","Shoot","Vascular","Nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8"))
```

```{r}
library(PMCMR)
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/TSS/rpkm_new/tissue/Tissue/CHH_all_2.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("root","shoot","vascular","nonvascular"))
ggplot(data=x, aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8")) + ylim(5,21)

```


Are these changes significant

```{r}
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/11_25_2015/dist_data/TSS_cs_plots/Tissue/CHH/CHH_2.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("Root","Shoot","Vascular","Nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8"))
```

```{r}
library(PMCMR)
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/TSS/rpkm_new/tissue/Tissue/CHH_all_3.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("root","shoot","vascular","nonvascular"))
ggplot(data=x, aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8")) + ylim(5,21)

```


Are these changes significant

```{r}
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/11_25_2015/dist_data/TSS_cs_plots/Tissue/CHH/CHH_3.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("Root","Shoot","Vascular","Nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8"))
```

```{r}
library(PMCMR)
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```


```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/TSS/rpkm_new/tissue/Tissue/CHH_all_4.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("root","shoot","vascular","nonvascular"))
ggplot(data=x, aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8")) + ylim(5,21)

```


Are these changes significant

```{r}
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```

```{r fig.width=7, fig.height=5}
file_name = "/Users/gturco/Documents/Data/Sorg/11_25_2015/dist_data/TSS_cs_plots/Tissue/CHH/CHH_4.txt"
x = read.table(file_name, col.names=c("pos","freq","mtype"))
x$mtype <- factor(x$mtype, levels = c("Root","Shoot","Vascular","Nonvascular"))
ggplot(data=x, 
       aes(x = pos, y = freq * 100 ,group=mtype, color = mtype)) +  geom_line(size = .3) +
  theme_classic() + theme(text=element_text(size=20))    +  theme(legend.position ="none", text=element_text(size=7), panel.margin = unit(0, "cm"), plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + xlab(NULL) + ylab("% Methylation") + scale_colour_manual(values=c("#c6141c","#fccb0c","#009987","#3261a8"))
```

```{r}
library(PMCMR)
kruskal.test(freq ~ mtype, data = x) 
posthoc.kruskal.nemenyi.test(freq ~ mtype, data=x, dist="Tukey")

```

```{r fig.width=8.5, fig.height=4}
file_name = "/Users/gturco/Documents/code/Brady/bs_seq_analysis/genic_analysis/sig_heatmaps/CHH_all.txt"
y = read.table(file_name ,sep='\t',header=TRUE)
colnames(y) <- c("pos","sig","mtype")
y$pval <- cut(y$sig, breaks = c(-Inf, 0.001, 0.005, 0.05, 1))
ggplot(y, aes(pos, mtype, fill = pval)) + geom_tile() + scale_fill_brewer(palette = "Greens",direction=-1)  +
  theme_classic() + theme(text=element_text(size=20))    +  
  theme(text=element_text(size=12), panel.margin = unit(0, "cm"), 
        plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), 
        axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + 
  xlab(NULL) + ylab(NULL) 
  ```
  
```{r fig.width=8.5, fig.height=4}

file_name = "/Users/gturco/Documents/code/Brady/bs_seq_analysis/genic_analysis/sig_heatmaps/strict/CHH_all.txt"
y = read.table(file_name ,sep='\t',header=TRUE)
colnames(y) <- c("pos","sig","mtype")
y$pval <- cut(y$sig, breaks = c(-Inf, 0.001, 0.005, 0.05, 1))
ggplot(y, aes(pos, mtype, fill = pval)) + geom_tile() + scale_fill_brewer(palette = "Greens",direction=-1)  +
  theme_classic() + theme(text=element_text(size=20))    +  
  theme(text=element_text(size=12), panel.margin = unit(0, "cm"), 
        plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), 
        axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + 
  xlab(NULL) + ylab(NULL) 
  ```

RPKM Data
==============
  
1. first and last RPKM often most signifcant diffrnce seen in methylation
2. Larger diffrences seen in CHH and CHG the CG


```{r fig.width=8.5, fig.height=4}
sig = c(1.545e-05,0.005462,0,0.5438,0.02841,0,0.1316,0.001449,0,0.007113,0.0005268,0)
mtype = c("CG","CHG","CHH","CG","CHG","CHH","CG","CHG","CHH", "CG","CHG","CHH")
ttype = c("1","1","1","2","2","2","3","3","3","4","4","4")
x <- cbind(sig,mtype,ttype)
x <- data.frame(x)
x$sig = as.numeric(as.character(x$sig))
x$pval <- cut(x$sig, breaks = c(-Inf, 0.001, 0.005, 0.05, 1))
ggplot(x, aes(mtype, ttype, fill = pval)) + geom_tile() + scale_fill_brewer(palette = "Blues",direction=1) + theme_classic() + theme(text=element_text(size=20))    +  
  theme(text=element_text(size=12), panel.margin = unit(0, "cm"), 
        plot.margin = unit(c(0, 0, 0.01, 0), "cm"),axis.ticks = element_line(size = 0.1), 
        axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + 
  xlab(NULL) + ylab(NULL) 
  ```

RPKM Strict Data
==============
```{r fig.width=8.5, fig.height=4}

sig = c(0.0006828,0.006235,0,0.6834,0.001617,0,0.2797,0.0005292,0,0.04867,0.008849,0)
mtype = c("CG","CHG","CHH","CG","CHG","CHH","CG","CHG","CHH", "CG","CHG","CHH")
ttype = c("1","1","1","2","2","2","3","3","3","4","4","4")
x <- cbind(sig,mtype,ttype)
x <- data.frame(x)
x$sig = as.numeric(as.character(x$sig))
x$pval <- cut(x$sig, breaks = c(-Inf, 0.001, 0.005, 0.05, 1))
ggplot(x, aes(mtype, ttype, fill = pval)) + geom_tile() + scale_fill_brewer(palette = "Blues",direction=1) + theme_classic() + theme(text=element_text(size=20))    +  
  theme(text=element_text(size=12), panel.margin = unit(0, "cm"), 
        plot.margin = unit(c(0, 0, 0.01, 0), "cm"),axis.ticks = element_line(size = 0.1), 
        axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + 
  xlab(NULL) + ylab(NULL) 
  ```

Tissue Data
==============
```{r fig.width=8.5, fig.height=4}

sig = c(0.007,0.15,0.0002,0.0275,0.6112,0.028,0.8,0.87,0,0.005,0.094,0)
mtype = c("CG","CHG","CHH","CG","CHG","CHH","CG","CHG","CHH", "CG","CHG","CHH")
ttype = c("vascular","vascular","vascular","nonvascular","nonvascular","nonvascular","root","root","root","shoot","shoot","shoot")
x <- cbind(sig,mtype,ttype)
x <- data.frame(x)
x$sig = as.numeric(as.character(x$sig))
x$pval <- cut(x$sig, breaks = c(-Inf, 0.001, 0.005, 0.05, 1))
ggplot(x, aes(mtype, ttype, fill = pval)) + geom_tile() + scale_fill_brewer(palette = "Blues",direction=1) + theme_classic() + theme(text=element_text(size=20))    +  
  theme(text=element_text(size=12), panel.margin = unit(0, "cm"), 
        plot.margin = unit(c(0, 0, 0.01, -0.1), "cm"),axis.ticks = element_line(size = 0.1), 
        axis.line = element_line(size = 0.1), axis.ticks.length = unit(0.05, "cm"), axis.ticks.margin =unit(0, "cm")) + 
  xlab(NULL) + ylab(NULL) 
  ```